controller/src/enhancements/spectators_list.rs

```diff
@@ -0,0 +1,166 @@
+
+        let local_player_controller = match local_player_controller {
+            Some(controller) => controller,
+            None => {
+                /* We're currently not connected */
+                return Ok(());
+            }
+        };
+
+        let observice_entity_handle = if local_player_controller.m_bPawnIsAlive()? {
+            local_player_controller.m_hPawn()?.get_entity_index()
+        } else {
+            return Ok(());
+        };
+
+        for entity_identity in ctx.cs2_entities.all_identities() {
+            if entity_identity.handle::<()>()?.get_entity_index() == observice_entity_handle {
+                /* current pawn we control/observe */
+                continue;
+            }
+
+            let entity_class = ctx
+                .class_name_cache
+                .lookup(&entity_identity.entity_class_info()?)?;
+
+            if !entity_class
+                .map(|name| *name == "CCSPlayerController")
+                .unwrap_or(false)
+            {
+                /* entity is not a player pawn */
+                continue;
+            }
+
+            let player_controller_ptr = entity_identity.entity_ptr::<CCSPlayerController>()?;
+            let player_controller = player_controller_ptr.read_schema()?;
+            let player_pawn_handle = player_controller.m_hPawn()?;
+            let current_player_pawn = ctx.cs2_entities.get_by_handle(&player_pawn_handle)?;
+
+            let player_pawn = if let Some(identity) = &current_player_pawn
+            {
+                identity.entity()?.reference_schema()?
+            } else {
+                continue;
+            };
+
+            let observer_services_ptr = player_pawn.m_pObserverServices();
+            let observer_services = observer_services_ptr?
+                .try_reference_schema()
+                .with_context(|| obfstr!("failed to read observer services").to_string())?;
+
+            let observer_target_handle = match observer_services {
+                Some(observer) => observer.m_hObserverTarget()?,
+                None => {
+                    continue;
+                }
+            };
+
+            let current_target = ctx.cs2_entities.get_by_handle(&observer_target_handle)?;
+
+            let observer_target = if let Some(identity) = &current_target
+            {
+                identity.entity()?.cast::<C_CSObserverPawn>().reference_schema()?
+            } else {
+                continue;
+            };
+
+            let target_controller_handle = observer_target.m_hController()?;
+            let target_current_controller = ctx.cs2_entities.get_by_handle(&target_controller_handle)?;
+
+            let target_controller = if let Some(identity) = &target_current_controller
+            {
+                identity.entity()?.reference_schema()?
+            } else {
+                continue;
+            };
+
+            if target_controller.m_hPawn()?.get_entity_index() != observice_entity_handle {
```

Comment:
You can just compare the `target_controller_handle` with the handle to `local_player_controller`.  
This avoids some extra reads within this loop :)

Reply:
Sorry, I don't understand how am I supposed to compare them.